feat(tests): add test coverage for cli commands with no repo config

MTarek165 opened the pull request:
closes: #10416

### Description
 This PR covers two scenarios of cli commands that don't have configs , both uses ApplicationContext.builder().deduceEnvironment(false).start() to ensure that no configs passed and ensure isolation.
1- Used Namespace Kv command (without params) that don't depend on configs , with no configs passed ,should run successfully as expected.
2- Used Create flow command at the same environment and it depends on configs (DB repository), with no configs passed, should give " No bean of type [io.kestra.core.repositories.FlowRepositoryInterface] exists" and fail as expected.

MTarek165 committed:
fix(core): Add warning logs for mismatched (Parent-Subflow) inputs for subflow plugin.

MTarek165 committed:
feat(tests): add test coverage for cli commands with no repository configurations

MTarek165 committed:
refactor(tests): enhance test assertions and namings

loicmathieu reviewed (changes requested):
As it's test related, please don't create a new command but use an existing command in your test

MTarek165 commented:
@loicmathieu I did as asked to create a dummy command but now do you want me to remove this new command and use existing one that don't depend on config at the first scenario, right?

MTarek165 committed:
refactor(tests): replace HelloCommand with Namespace KV helper in no-config tests

- Use Namespace KV command (with no params) to verify repo-independent CLI behavior
- Remove HelloCommand usage in NoConfigCommandTest
- Keep FlowCreateCommand case to assert repo-dependent commands fail without config

MTarek165 commented:
@loicmathieu done, I removed the added command and replaced with existing one "kestra namespace kv" that don't depend on configs, now both scenarios tested on real commands.

loicmathieu reviewed (approved):
LGTM, thanks a lot
